tests/Microsoft.Bot.Builder.AI.QnA.Tests/QnAMakerTests.cs

```diff
@@ -57,6 +57,38 @@ await CreateFlow(rootDialog)
             .StartTestAsync();
         }
 
+        [TestMethod]
+        public async Task QnAMakerAction_Test1()
+        {
+            TypeFactory.Configuration = new ConfigurationBuilder().Build();
+            var mockHttp = new MockHttpMessageHandler();
+            mockHttp.When(HttpMethod.Post, GetRequestUrl()).WithContent("{\"question\":\"Q11\",\"top\":1,\"strictFilters\":[],\"metadataBoost\":[],\"scoreThreshold\":0.3}")
+                .Respond("application/json", GetResponse("QnaMaker_TopNAnswer.json"));
+            mockHttp.When(HttpMethod.Post, GetTrainRequestUrl())
+                .Respond(HttpStatusCode.NoContent, "application/json", "{ }");
+            mockHttp.When(HttpMethod.Post, GetRequestUrl()).WithContent("{\"question\":\"Q12\",\"top\":1,\"strictFilters\":[],\"metadataBoost\":[],\"scoreThreshold\":0.3}")
+               .Respond("application/json", GetResponse("QnaMaker_ReturnsAnswer_WhenNoAnswerFoundInKb.json"));
+
+            var rootDialog = CreateQnAMakerActionDialog(mockHttp);
+
+            var suggestionList = new List<string> { "Q1", "Q2", "Q3" };
+            var suggestionActivity = CardHelper.GetSuggestionsCard(suggestionList, "Did you mean:", "None of the above.");
+
```

Comment:
HEre you have 2 flows in one test. Make separate tests with meaningful names. You can create a common method for the setup above.